### PR TITLE
Create and display pmtiles for snow coverage

### PR DESCRIPTION
**Pull request by dgboss**
Provided snow coverage data is available from the NSIDC, we now create a pmtiles layer and store it in our object storage at psu/pmtiles/snow/{yyyy-mm-dd}/snowCoverage{yyyymmdd.pmtiles}

Front-end implementation and expected behavior
- On initial load, and everytime the Date of Interest changes, we query the API to get the date of the most recently processed snow that is less than or equal to the Date of Interest. If a date is returned, this is used in constructing the URL for fetching pmtiles.
- A 'Snow Coverage' option was added to the legend which allows you to toggle the snow layer on/off. Since the date of our snow coverage doesn't necessarily align with the Date of Interest due to processing delays, the legend displays the date the snow layer represents.

# Test Links:
[Landing Page](https://wps-pr-3444.apps.silver.devops.gov.bc.ca/)
[MoreCast 2.0](https://wps-pr-3444.apps.silver.devops.gov.bc.ca/morecast-2)
[Percentile Calculator](https://wps-pr-3444.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-3444.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-3444.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3444.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3444.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3444.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3444.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: api/app/jobs/viirs_snow.py [hunk 1]
```diff
@@ -135,6 +159,49 @@ async def _save_clipped_snow_coverage_mosaic_to_s3(self, for_date: date, path: s
...
+        # Classify the data. Snow coverage in the source data is indicated by values in the range of 0-100. I'm using a range of 
+        # 10 - 100 to increase confidence. In the classified data 1 is assigned to snow covered pixels and all other pixels are 0.
+        classified = np.where((source_data > 10) & (source_data <= 100), 1, 0)
+        output_driver = gdal.GetDriverByName("GTiff")
+        classified_snow_path = os.path.join(path, BINARY_SNOW_COVERAGE_CLASSIFICATION_NAME)
+        classified_snow = output_driver.Create(classified_snow_path, xsize=source_band.XSize, ysize=source_band.YSize, bands=1, eType=gdal.GDT_Byte)
+        classified_snow.SetGeoTransform(source.GetGeoTransform())
+        classified_snow.SetProjection(source.GetProjection())
+        classified_snow_band = classified_snow.GetRasterBand(1)
+        # snow_mask_band.SetNoDataValue(0) ???
```
Comment (brettedw) [inline api/app/jobs/viirs_snow.py:177]: Did you mean to delete this line?

**Review by brettedw — APPROVED** [review]
Looks great, nice work!

File: api/app/jobs/viirs_snow.py (hunk 1 above)
Comment (dgboss) [reply api/app/jobs/viirs_snow.py:177]: Ah, yes, was doing some testing and this line is not required. Thanks!

File: api/app/routers/snow.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+@router.get('/most-recent-by-date/{for_date}', response_model=ProcessedSnowResponse | None)
+async def get_most_recent_by_date(for_date: date, _=Depends(authentication_required)):
+    """ Returns the most recent processed snow record before or equal to the provided date. """
+    logger.info('/snow/most-recent-by-date/')
+    tz_aware_datetime = vancouver_tz.localize(datetime.combine(for_date, datetime.min.time()))
+    async with get_async_read_session_scope() as session:
+        result = await get_most_recent_processed_snow_by_date(session, tz_aware_datetime)
+        if result is not None:
+            processed_snow = result[0]
```
Comment (conbrad) [inline api/app/routers/snow.py:29]: I think the query returns the `first` value from the possible list of results, will this index access work?

**Review by conbrad — APPROVED** [review]
Nice, I like that the generic polygonization functions were pulled out 😎

File: api/app/routers/snow.py (hunk 2 above)
Comment (dgboss) [reply api/app/routers/snow.py:29]: The query actually returns the `first` value as a tuple, so the indexing is necessary.

**Pull request merged**